### PR TITLE
Org user quota improvements

### LINKED ISSUE #6404: Improvements organization user quota slider UI
screen shot 2016-01-15 at 12 02 37

- Avoid calculations in the views using model attributes. Instead add methods to the own model to return them directly to the view. (See [feature envy](https://sourcemaking.com/refactoring/smells/feature-envy))
- Return flash message if the user sends the form with an invalid value in the quota input.

cc @viddo @xavijam

### PR DESCRIPTION
**Pull request by MariaCheca**
Closes #6404 

- Invalid input error message set, avoiding user to submit form.
- Fixed submit action.

untitled

CR @xavijam

### CODE REVIEW

**Comment by MariaCheca** [comment]
Tests are passing now, please review it @xavijam

**Comment by xavijam** [comment]
- [ ] Editing the user quota, if you focus on text input and press ENT button, "delete user dialog" appears.
- [ ] Creating a new user, quota slider should be enabled, setting the default quota for a new user and maximum the rest of organization quota.

File: lib/assets/javascripts/cartodb/organization/organization_user_form.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/* Organization user form */
+
+module.exports = cdb.core.View.extend({
+  el: '.js-organizationUserForm',
+  events: {
+    'submit': '_validateFormSubmit'
+  },
+
+  _validateFormSubmit: function(event) {
+    if (this.$('.js-userQuotaError').length == 0) {
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/organization/organization_user_form.js]: try to use === tripe comparision

File: app/views/admin/organization_users/_form.html.erb [hunk 2]
```diff
@@ -159,7 +158,7 @@
   <div class="FormAccount-footer">
     <% if !@user.new? && current_user.organization_owner? -%>
       <% if @user != current_user %>
-        <button href="<%= CartoDB.url(self, 'delete_organization_user', {id: @user.username}, current_user) %>" class="Button Button--secondary js-deleteAccount">
+        <button href="<%= CartoDB.url(self, 'delete_organization_user', {id: @user.username}, current_user) %>" type="button" class="Button Button--secondary js-deleteAccount">
```
Comment (xavijam) [inline app/views/admin/organization_users/_form.html.erb]: :+1:

File: app/views/admin/organization_users/_form.html.erb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 
-<%= form_for @user, url: (@user.new? ? CartoDB.url(self, 'create_organization_user', {}, current_user) : CartoDB.url(self, 'update_organization_user', {id: @user.username}, current_user)), :class =>'Form' do |f| %>
+<%= form_for @user, url: (@user.new? ? CartoDB.url(self, 'create_organization_user', {}, current_user) : CartoDB.url(self, 'update_organization_user', {id: @user.username}, current_user)), :class =>'Form', :html => {:class => 'js-organizationUserForm edit_user'} do |f| %>
```
Comment (xavijam) [inline app/views/admin/organization_users/_form.html.erb]: Where does "edit_user" come from?

Comment (xavijam) [inline app/views/admin/organization_users/_form.html.erb]: Double :class! caution.

File: lib/assets/javascripts/cartodb/organization/organization_user_form.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+/* Organization user form */
+
+module.exports = cdb.core.View.extend({
+  el: '.js-organizationUserForm',
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/organization/organization_user_form.js]: Remove this thing because it is unnecessary.

**Comment by MariaCheca** [comment]
retest

**Comment by MariaCheca** [comment]
please retest

**Comment by MariaCheca** [comment]
retest this please

**Comment by MariaCheca** [comment]
retest this please

**Comment by MariaCheca** [comment]
retest this please

**Comment by MariaCheca** [comment]
:sparkles:

**Pull request merged**